public/src/components/amounts/amounts.tsx

```diff
@@ -0,0 +1,238 @@
+import React from 'react';
+import update from 'immutability-helper';
+
+import {createStyles, Theme, withStyles, WithStyles} from '@material-ui/core/styles';
+import FormLabel from '@material-ui/core/FormLabel';
+import FormControl from '@material-ui/core/FormControl';
+import Button from '@material-ui/core/Button';
+import SaveIcon from '@material-ui/icons/Save';
+import RefreshIcon from '@material-ui/icons/Refresh';
+
+import {fetchSettings, saveSettings, SettingsType} from '../../utils/requests';
+import {ContributionType, Region} from '../../utils/models';
+import AmountInput from './amountInput';
+import Amount from './amount';
+
+export interface Amount {
+  value: string,
+  isDefault?: boolean
+}
+
+type Amounts = {
+  [f in ContributionType]: Amount[]
+}
+
+type AmountsRegions = {
+  [r in Region]: Amounts
+}
+
+interface DataFromServer {
+  value: AmountsRegions,
+  version: string,
+}
+
+function isRegion(s: string): s is Region {
+  return Object.values(Region).includes(s)
+}
+
+const emptyAmounts: Amounts = {
+  ONE_OFF: [],
+  MONTHLY: [],
+  ANNUAL: []
+};
+
+const styles = ({ palette, spacing, mixins }: Theme) => createStyles({
+  form: {
+    display: 'flex',
+    flexDirection: 'row'
+  },
+  button: {
+    marginRight: spacing.unit * 2,
+    marginBottom: spacing.unit * 2
+  },
+  buttons: {
+    marginTop: spacing.unit * 2,
+    marginLeft: spacing.unit * 4
+  },
+  regions: {
+    display: 'flex',
+    flexDirection: 'column'
+  },
+  region: {
+    paddingRight: spacing.unit * 4,
+    paddingLeft: spacing.unit * 4,
+    borderBottom: `1px solid ${palette.grey['300']}`,
+    marginBottom: spacing.unit * 4,
+    flexDirection: 'column'
+  },
+  ContributionType: {
+    margin: spacing.unit,
+    display: 'flex',
+    borderBottom: `1px dotted ${palette.grey['300']}`,
+    paddingBottom: '2px'
+  },
+  ContributionTypeName: {
+    fontWeight: 'bold',
+    marginTop: 'auto',
+    minWidth: '88px'
+  }
+});
+
+interface Props extends WithStyles<typeof styles> {}
+
+class AmountsComponent extends React.Component<Props, AmountsRegions> {
```

Comment:
I think we could rename these `...Component` ones across the board for clarity, since component is such a general term (everything is a component). Reading this through initially I was getting confused between what were the granular elements and higher-level parents - maybe `AmountsForm`, `ContributionTypesForm`, etc would be clearer?

Reply:
or maybe just the form ones, to `AmountsFormComponent`, `ContributionTypesFormComponent`?